### PR TITLE
Make Tomcat Access Log's fileDateFormat configurable via the environment

### PR DESCRIPTION
**Pull request by Nowheresly**
Fixes gh-8396

Please see issue #8396 for details.

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@Nowheresly Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=8474)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-boot&pullRequestId=8474) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@Nowheresly Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-boot&pullRequestId=8474)!

**Comment by Nowheresly** [comment]
If needed, I can provide the port to the master branch as well via another PR.

**Comment by philwebb** [comment]
Sweet. Thanks!. No need to create a different version, we'll port it to branches as required. I'm noted a couple of suggestions in the commit.

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java [hunk 1]
```diff
@@ -1071,6 +1072,13 @@ public void customize(Context context) {
 			 */
 			private boolean buffered = true;
 
+			/**
+			 * Customized date format in the access log file name. <a href=
+			 * "http://tomcat.apache.org/tomcat-7.0-doc/config/valve.html#Access_Log_Valve">Tomcat
```
Comment (snicoll) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java]: You cannot use HTML in field javadoc as we use that to generate description for the configuration key. Please look at other keys in that file for more inspiration.

File: spring-boot-docs/src/main/asciidoc/appendix-application-properties.adoc [hunk 2]
```diff
@@ -194,6 +194,7 @@ content into your application; rather pick only the properties that you need.
 	server.ssl.trust-store-provider= # Provider for the trust store.
 	server.ssl.trust-store-type= # Type of the trust store.
 	server.tomcat.accept-count= # Maximum queue length for incoming connection requests when all possible request processing threads are in use.
+	server.tomcat.accesslog.fileDateFormat=yyyy-MM-dd # Customized date format in the access log file name.
```
Comment (snicoll) [inline spring-boot-docs/src/main/asciidoc/appendix-application-properties.adoc]: Keys are ordered alphabetically and use a canonical name (`file-data-format`). The description must match the one in the field's Javadoc. Again, check other keys for consistency.

File: spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java (hunk 1 above)
Comment (Nowheresly) [inline spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/ServerProperties.java]: Maybe this link is not that useful

File: spring-boot-docs/src/main/asciidoc/appendix-application-properties.adoc (hunk 2 above)
Comment (Nowheresly) [inline spring-boot-docs/src/main/asciidoc/appendix-application-properties.adoc]: Thanks for your review. Makes sense indeed. Done.

**Comment by Nowheresly** [comment]
Unfortunately, I don't really understand why checkstyle is complaining in the Travis build...

**Comment by snicoll** [comment]
Thanks for your contribution. I've polished it in f8bf05b (in particular, the default value was missing a dot).

**Comment by Nowheresly** [comment]
Thanks for spotting the missing dot! and thanks as well for the time spent on reviewing my PR.

**Pull request closed without merging**